### PR TITLE
FEATURE: Separated references and properties in Neos UI

### PR DESCRIPTION
**Pull request by pKallert**
**What I did**
Currently, we are merging the references into the properties in the Neos UI. Since they are now separated into properties: and references: in the config, we should also handle them as such in the UI. 

The logic for the core is changed here: https://github.com/neos/neos-development-collection/pull/5156

### CODE REVIEW

**Review by dlubitz — APPROVED** [review]
LGTM!

File: packages/neos-ts-interfaces/src/index.ts [hunk 1]
```diff
@@ -142,7 +142,40 @@ export interface PropertyConfiguration {
...
+        inline?: {
+            editor?: string;
+            editorOptions?: {
+                [propName: string]: any;
+            };
+        }
+        inlineEditable?: boolean;
+        inspector?: {
+            hidden?: boolean;
+            defaultValue?: string;
```
Comment (mhsdesign) [inline packages/neos-ts-interfaces/src/index.ts]:
```suggestion
```

not legit options for references i think

File: packages/neos-ts-interfaces/src/index.ts [hunk 2]
```diff
@@ -142,7 +142,40 @@ export interface PropertyConfiguration {
...
+            position?: number | string;
+        };
+        help?: {
+            message?: string;
+            thumbnail?: string;
+        };
+    };
+    validation?: {
+        [propName: string]: ValidatorConfiguration | undefined;
+    };
```
Comment (mhsdesign) [inline packages/neos-ts-interfaces/src/index.ts]:
```suggestion
```

not legit options for references i think

File: packages/neos-ts-interfaces/src/index.ts [hunk 3]
```diff
@@ -142,7 +142,40 @@ export interface PropertyConfiguration {
...
+    properties?: {
```
Comment (mhsdesign) [inline packages/neos-ts-interfaces/src/index.ts]: is this a new feature ✨Native support in the Neos ui?? :D Hurray

File: packages/neos-ts-interfaces/src/index.ts [hunk 4]
```diff
@@ -142,7 +142,40 @@ export interface PropertyConfiguration {
...
+    ui?: {
+        label?: string;
+        reloadIfChanged?: boolean;
+        inline?: {
+            editor?: string;
+            editorOptions?: {
+                [propName: string]: any;
+            };
+        }
+        inlineEditable?: boolean;
```
Comment (mhsdesign) [inline packages/neos-ts-interfaces/src/index.ts]:
```suggestion
```

not legit options for references i think

File: packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts [hunk 5]
```diff
@@ -215,6 +218,20 @@ export default class NodeTypesRegistry extends SynchronousRegistry<NodeType> {
                                     helpThumbnail: property.ui?.help?.thumbnail
                                 })
                             ),
+                        ...references.filter(p => p.ui?.inspector?.group === group.id)
+                                .map(property => ({
```
Comment (mhsdesign) [inline packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts]:
```suggestion
                                .map(reference => ({
```

... and the rest :D

File: packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts [hunk 6]
```diff
@@ -215,6 +218,20 @@ export default class NodeTypesRegistry extends SynchronousRegistry<NodeType> {
...
+                                    type: 'editor',
+                                    id: property.id,
+                                    label: property.ui?.label,
+                                    editor: property.ui?.inspector?.editor,
+                                    editorOptions: property.ui?.inspector?.editorOptions,
+                                    position: property.ui?.inspector?.position,
+                                    hidden: property.ui?.inspector?.hidden,
+                                    helpMessage: property.ui?.help?.message,
+                                    helpThumbnail: property.ui?.help?.thumbnail,
+                                    properties: property?.properties
```
Comment (mhsdesign) [inline packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts]: ah we just expose it here but its not interpreted anywhere? I think we should use a better term here ...

**Review by mhsdesign — COMMENTED** [review]
Ah i see its indeed doable :D

File: packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts (hunk 6 above)
Comment (pKallert) [reply packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts]: Yes, would be a follow-up to use it in the editor, for now it is not used anywhere. 

Would perhaps edgeProperties be better?

File: packages/neos-ts-interfaces/src/index.ts (hunk 3 above)
Comment (pKallert) [reply packages/neos-ts-interfaces/src/index.ts]: Well, rather the preparation for using edge properties 😄

File: packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts (hunk 6 above)
Comment (mhsdesign) [reply packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts]: or `referenceProperties`? Maybe introduce this thing first with the mentioned followup and not now? ... on the other hand its just the yaml mapped 1to1 so we should probably not mess with it too much? ... but its not 1to1 really so maybe `referenceProperties`? :D

Comment (pKallert) [reply packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts]: `referenceProperties` works for me, it is also used fairly often in the documentation: https://docs.neos.io/guide/content-repository/nodetype/references-vs-properties 
I can also remove it for now, it does not really make a difference if we add it now or later.

Comment (mhsdesign) [reply packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts]: > it does not really make a difference if we add it now or later.

true in this case its simple ... but id do it from time to time to delay the discussion and let the reviewers focus more on the core parts :D ... Id vote for removing it now as without using this thing we dont know what the editor actually needs.
For example this pr would just pass the properties 1to1 but maybe they should also be mapped and get some view configuration? For people already wanting to experiment, there are alternative ways to fetch the reference properties by getting the js nodetype so i think that would be fine :D

Comment (pKallert) [reply packages/neos-ui-contentrepository/src/registry/NodeTypesRegistry.ts]: I removed it for now 👍

**Review by mhsdesign — APPROVED** [review]
Thanks looks perfect ;) ... but we should wait with merging for the neos change ofc

**Pull request merged**